### PR TITLE
Add support of GCF to `dev/start` script

### PR DESCRIPTION
**Pull request by nessiLahav**
### What does this PR do?
Add support of generating google cloud function tokens as part of `dev/start` script, 
in order to be able to run the automation tests in a local environmnet 

### What ticket does this PR close?
Connected to #1812

### CODE REVIEW

File: dev/cli [hunk 1]
```diff
@@ -83,6 +161,37 @@ _get_gcp_token_payload() {
...
+  local audience="dummy_host"
+  local status_code=$(curl -o /dev/null -s -w "%{http_code}\n" \
+    -H 'Metadata-Flavor: Google' "$gcp_func_url?audience=$audience")
+
+  if [ ! "$status_code" = "200" ]; then
+    echo "Failed to access function, HTTP Status: $status_code"
+    exit 1
+  fi
+}
+
```
Comment (eranha) [inline dev/cli:194]: source the original script and call the function.

File: dev/cli [hunk 2]
```diff
@@ -36,15 +36,93 @@ EOF
...
+  local option_arg=$2
+  if [ -z "$option_arg" ]; then
+    echo "argument is missing for $option option"
+    exit 1
+  fi
+}
+
+function print_gcp_option_error() {
+  local option=$1
+  if [ -z "$option" ]; then echo "required command options are missing";
```
Comment (eranha) [inline dev/cli:101]: Could you break the if?
```
if [ ]; then
else
fi
```

**Review by eranha — CHANGES_REQUESTED** [review]

File: dev/cli [hunk 3]
```diff
@@ -83,6 +161,37 @@ _get_gcp_token_payload() {
...
+  local gcp_func_url=$1
+  local token_prefix="tokens/gcf_"
+
+  validate_gcf_url_accessible $gcp_func_url
+
+  cd ../ci/authn-gcp/
+  sh ./get_tokens_to_files.sh "$gcp_func_url" "$token_prefix" || exit 1
+  cd -
+
+  echo "Finished obtaining and writing tokens to file"
```
[the remark this answers is not in the document]
Comment (nessiLahav) [reply dev/cli:178]: Yes, for development env is good enough

File: dev/cli (hunk 1 above)
Comment (nessiLahav) [reply dev/cli:188]: source is actually running the script, i extracted the logic to another script

Comment (eranha) [reply dev/cli:185]: because there's  a call to a function at the bottom of the script.

**Review by eranha — APPROVED** [review]

**Pull request merged**